plugins/greatexpectations/flytekitplugins/greatexpectations/task.py

```diff
@@ -0,0 +1,166 @@
+from great_expectations.exceptions import ValidationError
+
+from flytekit import PythonInstanceTask
+from flytekit.extend import Interface
+
+T = typing.TypeVar("T")
+
+
+@dataclass
+class BatchRequestConfig(object):
+    """
+    Use this configuration to configure Batch Request. A BatchRequest can either be
+    a simple BatchRequest or a RuntimeBatchRequest.
+
+    Args:
+        data_connector_query: query to request data batch
+        runtime_parameters: parameters to be passed at runtime
+        batch_identifiers: identifiers to identify the data batch
+        batch_spec_passthrough: reader method if your file doesn't have an extension
+    """
+
+    data_connector_query: typing.Optional[typing.Dict[str, typing.Any]] = None
+    runtime_parameters: typing.Optional[typing.Dict[str, str]] = None
+    batch_identifiers: typing.Optional[typing.Dict[str, str]] = None
+    batch_spec_passthrough: typing.Optional[typing.Dict[str, typing.Any]] = None
+
+
+class GETask(PythonInstanceTask[BatchRequestConfig]):
+    """
+    This task can be used to validate your data.
+
+    Args:
+        name: name of the task
+        data_source: tell where your data lives and how to get it
+        expectation_suite: suite which consists of the data expectations
+        data_connector: connector to identify data batches
+        inputs: inputs to pass to the execute() method
+        checkpoint_params: optional SimpleCheckpoint parameters
+        task_config: batchrequest config
+        data_context: directory in which GE's configuration resides
+
+    TODO: Connect Data Docs to Flyte Console.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        data_source: str,
+        expectation_suite: str,
+        data_connector: str,
+        inputs: typing.Dict[str, typing.Type],
+        checkpoint_params: typing.Optional[typing.Dict[str, typing.Union[str, typing.List[str]]]] = None,
+        task_config: BatchRequestConfig = None,
+        data_context: str = "./great_expectations",
+        **kwargs,
+    ) -> typing.Dict[str, str]:
+
+        _TASK_TYPE = "greatexpectations"
+        self._data_source = data_source
+        self._data_connector = data_connector
+        self._expectation_suite = expectation_suite
+        self._batch_request = task_config
+        self._data_context = data_context
+        self._checkpoint_params = checkpoint_params
+
+        super(GETask, self).__init__(
+            name=name,
+            task_config=task_config,
+            task_type=_TASK_TYPE,
+            interface=Interface(inputs=inputs),
+            **kwargs,
+        )
+
+    def execute(self, **kwargs) -> typing.Any:
+        context = ge.data_context.DataContext(self._data_context)
+
+        if len(self.python_interface.inputs.keys()) != 1:
+            raise RuntimeError(f"Expected one input argument to validate the dataset")
+
+        try:
+            dataset = kwargs[list(self.python_interface.inputs.keys())[0]]
```

Comment:
you don't have to do this check. this is done automatically by flyte base tasks

Reply:
also do you only want one argument?

Reply:
Yes, only one.

Reply:
is the argument always datasst

Reply:
Yes, it is.